contrib/check-whitespace.jl

```diff
@@ -0,0 +1,55 @@
+#!/usr/bin/env julia
+
+const patterns = split("""
+    *.1
+    *.c
+    *.cpp
+    *.h
+    *.inc
+    *.jl
+    *.lsp
+    *.make
+    *.md
+    *.mk
+    *.rst
+    *.scm
+    *.sh
+    *.yml
+    *Makefile
+""")
+
+const errors = Set{Tuple{String,Int,String}}()
+
+for path in eachline(`git ls-files -- $patterns`)
+    lineno = 0
+    non_blank = 0
+
+    file_err(msg) = push!(errors, (path, 0, msg))
+    line_err(msg) = push!(errors, (path, lineno, msg))
+
+    for line in eachline(path, keep=true)
+        lineno += 1
+        contains(line, '\r')   && file_err("non-UNIX line endings")
```

Comment:
Should this report the line number too? Maybe at least if it's not the first line, or do you think it's very unlikely for inconsistent newline characters to occur in the same file?

Reply:
If you include the line number then you get a separate error message for every single line of the file, which is generally unhelpful. This will happen if someone edited the file on Windows in an editor that converts the line endings to CRLF, which it will do for all the lines, not just the ones that were edited. And the line number isn't helpful for fixing it either since you  generally ask your editor to convert the line endings of the entire file to CR. There aren't any editors that let you save part of a file with CRLF and other parts of a file with CR line endings.

Reply:
Yeah, I guess that makes sense